src/sitemap/handler.js

```diff
@@ -276,6 +277,22 @@ export async function findSitemap(inputUrl) {
   );
   const extractedPaths = await getBaseUrlPagesFromSitemaps(inputUrl, filteredSitemapUrls);
 
+  // check if URLs from each sitemap exist and remove entries if none exist
+  if (Object.entries(extractedPaths).length > 0) {
+    const extractedSitemapUrls = Object.keys(extractedPaths);
+    for (const s of extractedSitemapUrls) {
+      const urlsToCheck = extractedPaths[s];
+      // eslint-disable-next-line no-await-in-loop
+      const existingPages = await checkCommonSitemapUrls(urlsToCheck);
```

Comment:
I think at this point we should rename this function, it does not really check common sitemap urls, but just make a head requests to check if the response is 200. What about `filterValidUrls`?

Reply:
yap, that's a good suggestion.